6.4.remember_remember.py

```diff
@@ -0,0 +1,30 @@
+# Author: Omer Gertler
+
+from PIL import Image
+
+DIR = r"C:\Users\omer\Desktop\Notebooks-master\Notebooks-master\week06\resources\code.png"
+
+
+def decipher_msg_from_pic(image_dir):
+    """
+    find all black pixels (numeric value == 1) in the image.
+    searching column by column left to right, each column top down.
+    for every black pixel, it's line-number translated to ascii value.
+    the result is a string message of those decipher line numbers.
+    :param image_dir: the directory where the image stores
+    :return: string of decipher line numbers of the black pixels
+    """
+    black_pixels = []
+    with Image.open(image_dir) as img:
```

Comment:
A good rule of thumb is that more than 3 tabs indicate that the code is not clean. You can use a few solutions to solve this problem:
1. Replace for loops -  
https://medium.com/python-pandemonium/never-write-for-loops-again-91a5a4c84baf
Replace for loops with more advanced python features 

2.Replace these two for loops with a list comprehention with two iterators - 
https://stackoverflow.com/questions/18551458/how-to-frame-two-for-loops-in-list-comprehension-python

3.  You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.